lib/kubernetes-deploy/restart_task.rb

```diff
@@ -112,7 +112,7 @@ def build_watchables(kubeclient_resources, started)
     def verify_namespace
       kubeclient.get_namespace(@namespace)
       @logger.info("Namespace #{@namespace} found in context #{@context}")
-    rescue KubeException => error
+    rescue Kubeclient::HttpError => error
```

Comment:
Might be out of scope of this PR but should we be differentiating between `HttpError` and `ResourceNotFoundError`

Reply:
In [Kubeclient](https://github.com/abonas/kubeclient/blob/master/lib/kubeclient/resource_not_found_error.rb), `ResourceNotFoundError` is a subclass of HttpError so we will be catching it regardless. From what I see, Kubeclient always provides response code with `HttpError`, so our current checks in kubernetes-deploy will still work all the same.

Also worth noting that `KubeException` is [going to be removed on next release](https://github.com/abonas/kubeclient/pull/293) so this is a requirement for future kubeclient updates.